### ISSUE TITLE
Magento 2.2.7 catalog price rule "conditions SKU is" not apply

### ISSUE THREAD

**Issue by tudn9x** [magento/magento2#21888]
### Preconditions
1.Magento 2.2.7

Steps to reproduce:

1. create a catalog rule on all customer group
1.1 conditions SKU is (select the three simple product)
1.2 Actions : adjust final price to discount value - discount amount 38.0000
1.3 discard subsequent rules : NO
2. disable all other rules
3. reindex

### Expected result
1. on front all item have a new price of 38 and old price with line-through

### Actual result
1. on front all item has old price nothing change

I tried with conditions SKU is one of, it working fine.

**Comment by magento-engcom-team** [comment]
Hi @htmyheart. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@htmyheart do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by tudn9x** [comment]
@magento-engcom-team yes

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by maddumarala** [comment]
@htmyheart 
1.1 conditions SKU is (select the three simple product)
Why do you put three simple product when the condition is "IS".
'SKU is' for singular use.You can use contains or is one of for multiple SKUs.

**Comment by ghost** [comment]
HI @htmyheart thank you for you report, this is non-issue please use  **is one of**

**Comment by tudn9x** [comment]
I'm sorry but please check and test before ask me any thing @maddumarala 

1. With condition SKU "is":
I created new rule: Change final price product with condition SKU "is" 24-MB01. But, nothing change!

2. With condition SKU "is one of" 24-MB04, 24-MB03, 24-MB05. Only first product with sku 24-MB04 change new price. Products 24-MB03, 24-MB05 nothing change.

3. With condition  SKU "containt" 24-MB02, 24-UB02, 24-WB01. All product both nothing change.

Please check again @engcom-backlog-nazar. This is not joke.

**Comment by ghost** [comment]
@htmyheart sorry, i'm misunderstood, if you select all three SKU, Issue is reproducible, if you select is one of and the list of products, then the price is set only for the first product from the list.

**Comment by tudn9x** [comment]
@engcom-backlog-nazar thank you very much. I'm sorry for my last message. I'm also sorry @maddumarala 

Ok, back to this issue, I only list 3 cases to make example. And you can compare case 1 and case 3 to see what issue is. And if you can, please check again and let me know if you need anything.

Thank you very much.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98856`, `MAGETWO-98857` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by arijit-sezzle** [comment]
@magento-engcom-team: How can I assign myself to work on this issue?

**Comment by arijit-sezzle** [comment]
I am working on this issue.

**Comment by arijit-de93** [comment]
@magento-engcom-team: How can I assign myself to work on this issue?

**Comment by magento-engcom-team** [comment]
Hi @arijit-de. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by vrajeshpatel4u** [comment]
Hi,
I have tested this Rules and applied 'is one of' and 'is' as a condition. In both the cases rules is not working. Products are not discounted. I have tried in fresh Magento. 

![Sku2](https://user-images.githubusercontent.com/40518060/55159872-0a329500-5188-11e9-8621-cc6f34cc1324.png)
![Products](https://user-images.githubusercontent.com/40518060/55159875-0a329500-5188-11e9-8647-7113add25180.png)
![SKURules](https://user-images.githubusercontent.com/40518060/55159876-0acb2b80-5188-11e9-96ea-28254970274a.png)
